packages/canvas-runtime-utils/src/utils.ts

```diff
@@ -162,70 +162,52 @@ export function getStylesAlias(id: string, componentTree: Tree, cssTree: Tree) {
 }
 
 export function createObject(
-  referenceObject: any,
-  keys: (string | number)[],
+  referenceObj: any,
+  selector: (string | number)[],
   value: string | number | boolean | string[] | number[] | boolean[] | object
 ) {
-  if (referenceObject === null) {
-    throw "Reference object cannot be null";
+  // Check if the object is null or undefined
+  if (referenceObj === null || referenceObj === undefined) {
+    throw new Error("Cannot patch a null or undefined object");
   }
-  if (keys.length === 0) {
+
+  // Create a copy of the object
+  const newObj = JSON.parse(JSON.stringify(referenceObj));
+
+  // Base case: if the selector array is empty, set the value of the new object to the given value
+  if (selector.length === 0) {
```

Comment:
It checks on this line if the selector array is empty it means we have traversed to the exact property that is to be patched so I return the value